### ISSUE TITLE
double divide gradient_accumulation_steps

### ISSUE THREAD

**Issue by efsotr** [huggingface/accelerate#3533]
### System Info

```Shell
newest version
```

### Information

- [ ] The official example scripts
- [x] My own modified scripts

### Tasks

- [ ] One of the scripts in the examples/ folder of Accelerate or an officially supported `no_trainer` script in the `examples` folder of the `transformers` repo (such as `run_no_trainer_glue.py`)
- [x] My own task or dataset (give details below)

### Reproduction

accelerator backward:
```python
def backward(self, loss, **kwargs):
        learning_rate = kwargs.get("learning_rate")

        if self.distributed_type != DistributedType.DEEPSPEED:
            # deepspeed handles loss scaling by gradient_accumulation_steps in its `backward`
            loss = loss / self.gradient_accumulation_steps
```
transformers trainer trainging_steps
```python
def training_step(...):
        ...
        if self.use_apex:
            with amp.scale_loss(loss, self.optimizer) as scaled_loss:
                scaled_loss.backward()
        else:
            # Finally we need to normalize the loss for reporting
            if not self.model_accepts_loss_kwargs and self.compute_loss_func is None:
                loss = loss / self.args.gradient_accumulation_steps

            # Turning off loss scaling w.r.t. gradient accumulation when DeepSpeed is enabled
            # https://github.com/huggingface/transformers/pull/35808
            if self.accelerator.distributed_type == DistributedType.DEEPSPEED:
                kwargs["scale_wrt_gas"] = False

            self.accelerator.backward(loss, **kwargs)

            return loss.detach()
```

### Expected behavior

no double divide

**Comment by SunMarc** [comment]
Hey @efsotr, we don't rely on accelerate anymore (we don't define `gradient_accumulation_steps` in accelerate ) to perform grad acc in transformers as the codebase is a lot more complicated with `num_item_in_batch` logic. Instead, we set `self.accelerator.gradient_state._set_sync_gradients(do_sync_step)` + we divide the loss by `gradient_accumulation_steps` by ourselves now 

LMK if this makes sense to you !
